### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by MarcCastellaPla**
Complete exercises and commented code

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+let mainTitle = document.getElementById("mainTitle");
+console.log(mainTitle)
+
+
+// Exercise 2: Changing HTML Content
+// TODO: Select all paragraph elements and change the content of the second paragraph to "This paragraph has been changed."
+
+let paragraph = document.getElementsByClassName("paragraph");
+paragraph[1].innerHTML = "<p>This paragraph has been changed.</p>";
```
Comment (tictools) [inline index.js]: Only text is required to change,not adding new HTML in the element.

- [x] Fix text content implementation.

File: index.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+// Select the element with the ID 'mainTitle' and log it to the console.
+
+let mainTitle = document.getElementById("mainTitle");
+console.log(mainTitle)
+
+
+// Exercise 2: Changing HTML Content
+// TODO: Select all paragraph elements and change the content of the second paragraph to "This paragraph has been changed."
+
+let paragraph = document.getElementsByClassName("paragraph");
```
Comment (tictools) [inline index.js]:
```suggestion
let paragraphsList = document.getElementsByClassName("paragraph");
```

File: index.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+let content = document.getElementById("content");
+content.style.backgroundColor = "lightgreen";
+
+// Exercise 4: Adding/Removing Classes
+// TODO: Add the "highlight" class to the first paragraph
+// TODO: Remove the "highlight" class from the first paragraph after 3 seconds
+
+paragraph[0].classList.add("highlight");
+setTimeout(function () { // This function adds 3 seconds delay to removing the class "highlight"
```
Comment (tictools) [inline index.js]: - [x] Remove comment

File: index.js [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    paragraph[0].classList.remove("highlight");
+}, 3000);
+
+ // EXERCISE 5: Creating and Appending Elements
+ // Create a new <button> element with the text 'Click Me'.
+ // Append this button to the element with the ID 'content'.
+
+ let button = document.createElement("button");
+ button.textContent = "Click Me";
+ document.getElementById("content").appendChild(button); // Adding a button at the end of the element with id content.
```
Comment (tictools) [inline index.js]: - [ ] Remove comment

File: index.js [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+
+ let result = document.getElementById("result");
+ result.remove();
+
+ // EXERCISE 7: Handling Events
+ // Add a 'click' event listener to the button created in Exercise 5.
+ // When clicked, the button should alert "Button Clicked!".
+
+ button.addEventListener("click", function () {
+
```
Comment (tictools) [inline index.js]: - [ ] Remove extra line

File: index.js [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+    result.innerText = input.value; // This doesn't work because the result div has been eliminated in exercice 6
+})
+
+ // Exercise 9: Toggling Visibility
+ // Add a click event listener to the button with ID 'actionButton' that toggles the visibility of the content div.
+
+let actionButton = document.getElementById("actionButton");
+actionButton.addEventListener("click", function() {
+
```
Comment (tictools) [inline index.js]: - [ ] Remove extra line

File: index.js [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+    content.classList.add("hidden");
```
Comment (tictools) [inline index.js]: This logic does not act as a toggle.
It should work following this logic:
```bash
if element is visible then remove class visible add class hidden
else remove class hidden and a add class visible
```

- [x] Fix logic

File: index.js [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+
+})
+
+ // Exercise 10: Modifying Attributes
+ // Select the input field and change its placeholder text to "Enter your name".
+
+input.placeholder = "Marc";
```
Comment (tictools) [inline index.js]: Placeholder text should be _"Enter your name"_, not your name. 😅

- [ ] Fix text

**Review by tictools — CHANGES_REQUESTED** [review]

File: index.js (hunk 3 above)
Comment (MarcCastellaPla) [reply index.js]: In the rubric, at the code quality section, you specify that there must be comments explaining key parts of the code, and that (between others) if there are no comments, the score is worse.

File: index.js (hunk 5 above)
Comment (MarcCastellaPla) [reply index.js]: Do I have to remove all the blank lines inside the functions, or only the 48?

Comment (tictools) [reply index.js]: Only the ones marked in the tasks, not all of them.

File: index.js (hunk 3 above)
Comment (tictools) [reply index.js]: Sure, you are right. Keep them, but take in account that in the future it will not be necessary anymore.

**Review by tictools — APPROVED** [review]

**Pull request merged**